packages/core/src/util/flux.ts

```diff
@@ -0,0 +1,179 @@
+/*
```

Comment:
Is there any doc that better explains how this magic works? Where did you get it?

Reply:
check this link in the comment ;) https://github.com/influxdata/influxdb/issues/16109#issuecomment-561697800
this + some reverse engineering of the flux AST via https://v2.docs.influxdata.com/v2.0/api/#operation/PostQueryAst

Reply:
"extern" is basically like a automatically imported file into the query, but we have to pass the flux AST insteadof source text

Reply:
Good job on your side ... still, there might be problems with backward compatibility, once we introduce something that does not seem to be stable enough. There is no documentation and the things that are reverse-engineered might change easily.  I am also not sure whether this is working against InfluxDB 1.8, which must be supported as well. We should also plan support for query parameters in all client libraries once it is stable, so they are in sync. I am now not sure whether this is a good step forward, maybe that we will need other stakeholders herein.

2 PRs would play better herein. The observable part of this PR is fine with no side-effects and ready to be merged soon. The variable part would require some argumentation to proceed.